File: Delete a node (by position, 1-based index).py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def delete_node(head, position):
    if head is None:
        return None
    if position == 1:
        return head.next

    temp = head
    for _ in range(position-2):
        if temp is None or temp.next is None:
            return head
        temp = temp.next
    if temp.next:
        temp.next = temp.next.next
    return head

File: Delete a node (by position, 1-based index).c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->next = NULL;
    return node;
}
struct Node* deleteNode(struct Node* head, int pos) {
    if (!head) return NULL;
    if (pos == 1) {
        struct Node* temp = head->next;
        free(head);
        return temp;
    }
    struct Node* curr = head;
    for (int i = 1; curr && i < pos-1; i++) curr = curr->next;
    if (curr && curr->next) {
        struct Node* temp = curr->next;
        curr->next = temp->next;
        free(temp);
    }
    return head;
}
